include/rocksdb/utilities/transaction_db.h

```diff
@@ -222,6 +222,16 @@ struct TransactionDBOptions {
   // pending writes into the database. A value of 0 or less means no limit.
   int64_t default_write_batch_flush_threshold = 0;
 
+  // This option is valid only for write-prepared/write-unprepared. Transaction
+  // will rely on this callback to determine if a key should be rolled back
+  // with Delete or SingleDelete when necessary. If the callback returns true,
+  // then SingleDelete should be used. If the callback is not callable or the
+  // callback returns false, then a Delete is used.
+  // The callback should not throw because RocksDB is not exception-safe.
```

Comment:
It might make sense to clarify the thread safety requirements as well.

Reply:
Good point. Forgot to do so, will add.